src/orion/algo/random.py

```diff
@@ -62,4 +64,11 @@ def observe(self, points, results):
 
         A simple random sampler though does not take anything into account.
         """
-        pass
+        self.alreadyrun += len(points)
```

Comment:
This would also lead to the issue if there is more workers than sample capacity. We could add it in suggest instead but then we do not know if the sample is valid and will be kept (potential duplicate due to race-conditions). By moving the check of is_done to Experiment.is_done we avoid these issues.

Reply:
Yes, do this in suggestion will have the duplicate issue.  As for more workers than sample capacity, my understanding is: each worker will be able to observe that global completed points and some of them may propose duplicated samples, but according to the producer logic, after backoff, the is_done logic should be True at each workers.

Reply:
Yes, mostly, but this depends on the parallel strategy. If there is none, then non-completed trials are not passed to observe and the algo will exhaust the search space.